Typing: Unexpected keyword argument `kind` for `isin`

### Describe the issue:

The implementation of `isin` has a `kind` argument
https://github.com/numpy/numpy/blob/cc91b105b983767eee194b9c4a348c1d3379c924/numpy/lib/arraysetops.py#L767-L768

that is missing in the type hints:
https://github.com/numpy/numpy/blob/cc91b105b983767eee194b9c4a348c1d3379c924/numpy/lib/arraysetops.pyi#L331-L336

### Reproduce the code example:

```python
import numpy as np

np.isin([1, 2, 3], [1], kind="table")
```


### Error message:

```shell
# mypy bug.py
bug.py:3: error: Unexpected keyword argument "kind" for "isin"  [call-arg]
bug.py:3: note: See https://mypy.rtfd.io/en/stable/_refs.html#code-call-arg for more info
Found 1 error in 1 file (checked 1 source file)
```


### Runtime information:

numpy 1.25.2
mypy 1.5.1 (compiled: yes)
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]


### Context for the issue:

_No response_

--- a/numpy/lib/arraysetops.pyi
+++ b/numpy/lib/arraysetops.pyi
@@ -333,6 +333,8 @@ def isin(
     test_elements: ArrayLike,
     assume_unique: bool = ...,
     invert: bool = ...,
+    *,
+    kind: None | str = ...,
 ) -> NDArray[bool_]: ...
 
 @overload

